makeflow/src/dag.h

```diff
@@ -19,6 +19,8 @@ See the file COPYING for details.
 
 #include <stdio.h>
 
+#define ARCHIVING_DEFAULT_DIRECTORY "/tmp/makeflow.archive."
```

Comment:
It seems odd to put this here, since it's more of a property of the execution than of the dag state itself.
Please call it `MAKEFLOW_ARCHIVE_DEFAULT_DIRECTORY` and put it in `makeflow_archive.h`